apstools/devices/aps_undulator.py

```diff
@@ -178,6 +183,54 @@ class STI_Undulator(PlanarUndulator):
     """
 
 
+class Undulator2M(ID_Spectrum_Mixin, ID_Controls_Mixin, ID_Misc_Mixin, Device):
+    """APS 2M Undulator.
+
+    .. index::
+        Ophyd Device; PlanarUndulator
+        Ophyd Device; Undulator2M
+
+    APS Use: 1ID, downstream.
+
+    EXAMPLE::
+
+        undulator = Undulator2M("S01ID:DSID:", name="undulator")
+    """
+
+    # PVs not found
+    busy = None
+    magnet = None
+    version_plc = None
+    version_hpmu = None
+
+    done = Component(EpicsSignalRO, "BusyM.VAL", kind="omitted")
+    done_value = 0
+
+
+class Undulator4M(ID_Spectrum_Mixin, ID_Controls_Mixin, ID_Misc_Mixin, Device):
+    """APS 4M Undulator.
+
+    .. index::
+        Ophyd Device; PlanarUndulator
+        Ophyd Device; Undulator4M
+
+    APS Use: 11ID, downstream & upstream.
+
+    EXAMPLE::
+
+        undulator = Undulator4M("S11ID:DSID:", name="undulator")
+    """
+
+    # PVs not found
+    busy = None
+    magnet = None
+    version_plc = None
```

Comment:
Isn't this the same content with just a different class name?

Reply:
Made 4M as a subclass of 2M